cloudinit/distros/ubuntu.py

```diff
@@ -41,5 +64,37 @@ def preferred_ntp_clients(self):
             self._preferred_ntp_clients = copy.deepcopy(PREFERRED_NTP_CLIENTS)
         return self._preferred_ntp_clients
 
+    def install_snap_packages(self, pkglist):
+        for pkg in pkglist:
+            subp.subp(["snap", "install", pkg])
+
+    def install_packages(self, pkglist):
+        """Install packages from either apt or snap.
+
+        We prefer apt here as to not unexpectedly install a snap when
+        a user was previously using an apt installed package. This will result
+        in a snap install needing to wait for an apt update. If you know your
+        package is a snap, call "install_snap_packages" directly.
+        """
+        if not subp.which("snap"):
+            return super().install_packages(pkglist)
```

Comment:
This should be installing "snap" from apt right? not pkglist?
```suggestion
            super().install_packages(["snap"])
```